### PR TITLE
[SPARK-41658][CONNECT][TESTS] Enable doctests in pyspark.sql.connect.functions

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
This PR proposes to enable doctests in pyspark.sql.connect.functions that is virtually the same as pyspark.sql.functions.

### Why are the changes needed?
To make sure on the PySpark compatibility and test coverage.

### Does this PR introduce any user-facing change?
No, doctest's only.

### How was this patch tested?
New Doctests Added

### CODE REVIEW

File: python/pyspark/sql/connect/functions.py [hunk 1]
```diff
@@ -2317,3 +2318,141 @@ def unwrap_udt(col: "ColumnOrName") -> Column:
...
+
+        # TODO(SPARK-41834): implement Dataframe.conf
+        del pyspark.sql.connect.functions.from_unixtime.__doc__
+        del pyspark.sql.connect.functions.timestamp_seconds.__doc__
+        del pyspark.sql.connect.functions.unix_timestamp.__doc__
+
+        # TODO(SPARK-41757): Fix String representation for Column class
+        del pyspark.sql.connect.functions.col.__doc__
+
+        # TODO: support data type: Timestamp(NANOSECOND, null)
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/functions.py:2382]: JIRA?

File: python/pyspark/sql/connect/functions.py [hunk 2]
```diff
@@ -2317,3 +2318,141 @@ def unwrap_udt(col: "ColumnOrName") -> Column:
...
+        del pyspark.sql.connect.functions.monotonically_increasing_id.__doc__
+
+        # TODO(SPARK-41840): Fix 'Column' object is not callable
+        del pyspark.sql.connect.functions.first.__doc__
+        del pyspark.sql.connect.functions.last.__doc__
+        del pyspark.sql.connect.functions.max_by.__doc__
+        del pyspark.sql.connect.functions.median.__doc__
+        del pyspark.sql.connect.functions.min_by.__doc__
+
+        del pyspark.sql.connect.functions.broadcast.__doc__
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/functions.py:2413]: Were they all also because of SPARK-41840?

File: python/pyspark/sql/functions.py [hunk 3]
```diff
@@ -3208,7 +3208,7 @@ def monotonically_increasing_id() -> Column:
 
     Examples
     --------
-    >>> df0 = sc.parallelize(range(2), 2).mapPartitions(lambda x: [(1,), (2,), (3,)]).toDF(['col1'])
+    >>> df0 = spark.sparkContext.parallelize(range(2), 2).mapPartitions(lambda x: [(1,), (2,), (3,)]).toDF(['col1'])
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py:3211]: Let's remove this change back for now.

**Comment by HyukjinKwon** [comment]
Thanks for working on this. Looks pretty good otherwise.

File: python/pyspark/sql/connect/functions.py (hunk 1 above)
Comment (techaddict) [reply python/pyspark/sql/connect/functions.py]: added the jira

File: python/pyspark/sql/connect/functions.py (hunk 2 above)
Comment (techaddict) [reply python/pyspark/sql/connect/functions.py:2421]: updated the rest of jiras

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
All related tests passed.

Merged to master.

Thanks for finishing this one @techaddict

**Pull request closed without merging**